### ISSUE TITLE
4.0.0-beta is missing @types dependencies

### ISSUE THREAD

**Issue by arcanis** [algolia/algoliasearch-client-javascript#975]
**Do you want to request a *feature* or report a *bug*?**

Rather a bug.

**What is the current behavior?**

Some packages don't depend on their @types dependencies, causing TS to report typecheck issues (in particular on strict package managers such as Yarn 2.x).

cc @Haroenv 

**If the current behavior is a bug, please provide all the steps to reproduce and a minimal
[JSFiddle](https://jsfiddle.net/) example or a repository on GitHub that we can `npm install`
and `npm start`.**

```
git clone git@github.com:yarnpkg/berry && cd berry
yarn typecheck:all
```

**What is the expected behavior?**

Shoudn't have errors, but instead we have:

```
.yarn/cache/@algolia-client-analytics-npm-4.0.0-beta.14-673c8771ce-1.zip/node_modules/@algolia/client-analytics/dist/client-analytics.d.ts:4:31 - error TS2307: Cannot find module '@algolia/client-search'.

4 import { SearchOptions } from '@algolia/client-search';
                                ~~~~~~~~~~~~~~~~~~~~~~~~

.yarn/cache/@algolia-client-common-npm-4.0.0-beta.14-cc4fb0d283-1.zip/node_modules/@algolia/client-common/dist/client-common.d.ts:1:25 - error TS2307: Cannot find module '@algolia/transporter'.

1 import { Headers } from '@algolia/transporter';
                          ~~~~~~~~~~~~~~~~~~~~~~

.yarn/cache/@algolia-client-common-npm-4.0.0-beta.14-cc4fb0d283-1.zip/node_modules/@algolia/client-common/dist/client-common.d.ts:2:29 - error TS2307: Cannot find module '@algolia/transporter'.

2 import { HostOptions } from '@algolia/transporter';
                              ~~~~~~~~~~~~~~~~~~~~~~

.yarn/cache/@algolia-client-common-npm-4.0.0-beta.14-cc4fb0d283-1.zip/node_modules/@algolia/client-common/dist/client-common.d.ts:3:33 - error TS2307: Cannot find module '@algolia/transporter'.

3 import { QueryParameters } from '@algolia/transporter';
                                  ~~~~~~~~~~~~~~~~~~~~~~

.yarn/cache/@algolia-client-common-npm-4.0.0-beta.14-cc4fb0d283-1.zip/node_modules/@algolia/client-common/dist/client-common.d.ts:4:32 - error TS2307: Cannot find module '@algolia/transporter'.

4 import { RequestOptions } from '@algolia/transporter';
                                 ~~~~~~~~~~~~~~~~~~~~~~

.yarn/cache/@algolia-client-common-npm-4.0.0-beta.14-cc4fb0d283-1.zip/node_modules/@algolia/client-common/dist/client-common.d.ts:5:29 - error TS2307: Cannot find module '@algolia/transporter'.

5 import { Transporter } from '@algolia/transporter';
                              ~~~~~~~~~~~~~~~~~~~~~~

.yarn/cache/@algolia-client-common-npm-4.0.0-beta.14-cc4fb0d283-1.zip/node_modules/@algolia/client-common/dist/client-common.d.ts:6:36 - error TS2307: Cannot find module '@algolia/transporter'.

6 import { TransporterOptions } from '@algolia/transporter';
                                     ~~~~~~~~~~~~~~~~~~~~~~


Found 7 errors.
```

**Comment by nunomaduro** [comment]
@arcanis This is indeed a bug. It will be fixed tomorrow.

**Comment by nunomaduro** [comment]
@arcanis This problem was been fixed on the version `4.0.0-beta.16`.
